src/Credit/Credit.cpp

```diff
@@ -66,7 +65,7 @@ struct CreditWidget : ModuleWidget {
                 saving.store(! saving.load());
 
                 if(saving.load())
-                    saveModules(mw);
+                    file_utils::saveToChosenFilePath([this] (const char* path) { saveModules(path, mw); });
```

Comment:
this won't work in async mode because `this` will get out of scope by the time the file browser action finishes (the menu is long gone by then).
instead we need to store the variables needed locally and pass them to the function.

this should work:

```
ModuleWriter& mw = this->mw;
file_utils::saveToChosenFilePath([mw] (const char* path) { saveModules(path, mw); });
```

Reply:
I'm not sure I understand... the `this` in the capture list is a `SaveButton`, which should have the same lifetime as the `CreditWidget`. If `this` has gone out of scope, then the `CreditWidget` has been deleted, and the `ModuleWriter` along with it.

Reply:
oh ok, I misunderstood then.
Was thinking of a menu action, those from right-click menus. nevermind this then :sweat_smile:

Reply:
All good!